src/meta/client/MetaClient.cpp

```diff
@@ -438,12 +454,12 @@ folly::Future<StatusOr<bool>> MetaClient::addHosts(const std::vector<HostAddr>&
                 }, true);
 }
 
-folly::Future<StatusOr<std::vector<HostAddr>>> MetaClient::listHosts() {
+folly::Future<StatusOr<std::vector<HostStatus>>> MetaClient::listHosts() {
     cpp2::ListHostsReq req;
     return getResponse(std::move(req), [] (auto client, auto request) {
                 return client->future_listHosts(request);
             }, [this] (cpp2::ListHostsResp&& resp) -> decltype(auto) {
-                return this->to(resp.hosts);
+                return this->toHostStatus(resp.hosts);
             });
```

Comment:
Be careful here! 
Now we need to list hosts with status,  but as current implementation,  the activeHostManager only valid on leader.  So we need to send the request to leader or we save the status in rocksdb

Reply:
If we save the status in rocksdb, we need to get host info from rocksdb when instantiate ActiveHostsMan, am I right?